wtf-compiler/examples/record_downcasting.wtf

```diff
@@ -0,0 +1,91 @@
+package wtf:example@0.1.0
+
+// Record type definitions for demonstrating downcasting
+record point2d {
+    x: s32,
+    y: s32
+}
+
+record person {
+    name: string,
+    age: s32
+}
+
+// Example function demonstrating record downcasting
+export func main() -> s64 {
+    // Simple case: Create records with exact matching fields
+    let point_2d_data = {
+        x: 1,
+        y: 2
+    }
+    
+    // This works - exact field match
+    let point_2d: point2d = point_2d_data
+    
+    return 0
+}
+
+// Tests demonstrating record downcasting behavior
+test "basic record downcasting from anonymous record to named type" {
+    let anonymous_record = {
+        x: 5,
+        y: 7
+    }
+    
+    let typed_point: point2d = anonymous_record
+    // Assert that the fields are accessible and have expected total
+    // This demonstrates that record downcasting preserves field values
+    assert typed_point.x + typed_point.y == 12
```

Comment:
assert fields individually, don't add them. Use as many asserts as you think make sense. Same below and for all tests in this file